src/stores/ViewportStore.ts

```diff
@@ -35,21 +34,24 @@ export default class ViewportStore extends Store<ViewportStoreState> {
     private readonly routeStore: RouteStore
     private readonly isSmallScreenQuery: () => boolean
 
-    constructor(routeStore: RouteStore, isSmallScreenQuery: () => boolean) {
+    constructor(routeStore: RouteStore, initialBBox: Bbox | null, isSmallScreenQuery: () => boolean) {
         super()
         this.routeStore = routeStore
         this.isSmallScreenQuery = isSmallScreenQuery
+        if (initialBBox) {
+            const updatedState = calculateLatLngFromBbox(this.state, initialBBox, isSmallScreenQuery())
+            super.setState(updatedState)
+        }
     }
 
     protected getInitialState(): ViewportStoreState {
         return {
-            // todo: initial values do not really matter, because we immediately update the state?!..
-            width: 800,
-            height: 600,
-            longitude: 11,
-            latitude: 48,
-            zoom: 5,
-        }
+            width: window.screen.width,
+            height: window.screen.height,
```

Comment:
Using the `window.screen.height/width` here is useful to get a rough estimation of the bounding box size, but maybe it could be more precise. The windows size is not equal to the map size and I think this is the reason why we still see a small flickering of the map: First the map is zoomed such that the markers are right at the map borders and then zoomed out a bit more. @Janekdererste do you have an idea how we could improve this?

Reply:
I think this is a reasonable heuristic, since the map is pretty much the same size as the window. This gets called before the first render pass and before any css is applied though. Therefore, it is not really possible to guess the map size correctly